### ISSUE TITLE
Bug: Cannot get session data after server redirecting

### ISSUE THREAD

**Issue by yusuke-shina** [codeigniter4/CodeIgniter4#3503]
**Describe the bug**
We cannot get the session value input and saved on the previous page with post type.
ex)[Page Transition Image]Sign in Page -> (POST) -> Validation Check -> (Server Redirecting) -> My Page

```php:MyPage Controller extends Base Controller
	public function index()
	{
		var_dump($this->session->get('user'));  # <- The value 'user' does not exist in the sessions. 
	}
```
I loaded the Session Library in the method `initController()` of the class `BaseController` as the name `$this->session` below.
```php:Base Controller
	public function initController(\CodeIgniter\HTTP\RequestInterface $request, \CodeIgniter\HTTP\ResponseInterface $response, \Psr\Log\LoggerInterface $logger)
	{
		// Do Not Edit This Line
		parent::initController($request, $response, $logger);
		//--------------------------------------------------------------------
		// Preload any models, libraries, etc, here.
		//--------------------------------------------------------------------
		// E.g.:
		// $this->session = \Config\Services::session();
		$this->session 		= \Config\Services::session();
		...
```

**CodeIgniter 4 version**
Version 4.0.4

**Affected module(s)**
Session Library + ?

**Expected behavior, and steps to reproduce if appropriate**
This shows the procedure in the Sign in flow I made.
1.  input the form items and post
2. save the values as the session data with `$this->session->set('user', 'xxx')` and redirect my page controller if succeed the form validation check.
3. get the values of the session data `$this->session->get('user')` in the my page controller, but it returns `NULL`. I expect it return 'xxx'.

**Context**
 - OS: [e.g. Windows 99] Alpine Linux( in the Docker Container) Host Machine : Mac OS version 10.15.6(19G73)
 - Web server [e.g. Apache 1.2.3] Nginx 1.19.1
 - PHP version [e.g. 6.5.4] PHP 7.4.8

**Comment by michalsn** [comment]
I was not able to reproduce this issue. I will assume that your session is created correctly so it's probably something with your code logic.

I'm closing this, since it doesn't seem to be CI fault, but feel free to continue conversation. Especially if you can provide a quick code example to reproduce the issue.

**Comment by yusuke-shina** [comment]
@michalsn
Thank you for your confirmation, and I'm sorry for the lack of confirmation.
I check my code logic, again.

**Comment by qvsinh** [comment]
Hi guys, I got this issue again. here are steps to reproduce this issue:
I used mysql database to store session.
1. Go to home page
2. access to other controller and create a session then redirect to other site (payment example), fill data then redirect back to our website (callback url). we get session here and empty.
I checked in ci_sessions table and got 2 rows. Before redirect to outside we just have 1 row.

I hope this issue can be fixed.
Thanks
